pages/index.tsx

```diff
@@ -1,72 +1,103 @@
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+import { Button } from "../components/Button";
+import {Form} from "../components/Form";
+import { IResponse } from "../interface/IResponse";
+import { SendForm } from "../api/sendForm";
+import styles from "../styles/Home.module.css";
+import { Heading } from "../components/Heading";
+import { SubmitMessage } from "../components/SubmitMessage";
+import { theme } from "../theme";
 
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
+const Home: NextPage = () => {
+  const [email, setEmail] = useState<string>("");
+  const [responseData, setResponseData] = useState<IResponse | null>(null);
+  const [emailValid, setEmailValid] = useState<boolean>(true);
 
-        <p className={styles.description}>
-          Get started by editing{' '}
-          <code className={styles.code}>pages/index.tsx</code>
-        </p>
+  const sendForm = async () => {
+    const res = await new SendForm(email).send({ body: { email: email } });
+    setResponseData(res);
+  };
 
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h2>Documentation &rarr;</h2>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+  const validateEmail = (email: string) => {
+    const regex = /[a-z0-9]+@[a-z]+\.[a-z]{2,3}/;
+    regex.test(email) ? setEmailValid(true) : setEmailValid(false);
+    return regex.test(email);
+  };
 
-          <a href="https://nextjs.org/learn" className={styles.card}>
-            <h2>Learn &rarr;</h2>
-            <p>Learn about Next.js in an interactive course with quizzes!</p>
-          </a>
+  const onInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail((e.target as HTMLInputElement).value);
+  };
 
-          <a
-            href="https://github.com/vercel/next.js/tree/canary/examples"
-            className={styles.card}
-          >
-            <h2>Examples &rarr;</h2>
-            <p>Discover and deploy boilerplate example Next.js projects.</p>
-          </a>
+  const onEnterPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      e.preventDefault();
+      if (validateEmail(email)) {
+        sendForm();
+      }
+    }
+  };
 
-          <a
-            href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
-            className={styles.card}
-          >
-            <h2>Deploy &rarr;</h2>
-            <p>
-              Instantly deploy your Next.js site to a public URL with Vercel.
-            </p>
-          </a>
-        </div>
-      </main>
+  const onSubmit = (
+    e: React.MouseEvent<HTMLButtonElement> | React.FormEvent<HTMLFormElement>
+  ) => {
+    e.preventDefault();
+    if (validateEmail(email)) {
+      sendForm();
+    }
```

Comment:
Duplicated lines with `37-41`. It will be good idea to extract it to separate function.